lighthouse-core/audits/seo/is-crawlable.js

```diff
@@ -0,0 +1,109 @@
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit');
+const BLOCKLIST = new Set([
+  'noindex',
+  'none',
+]);
+const ROBOTS_HEADER = 'x-robots-tag';
+const UNAVAILABLE_AFTER = 'unavailable_after';
+
+/**
+ * Checks if given directive is a valid unavailable_after directive and if content expired
+ * @param {string} directive
+ * @returns {boolean}
+ */
+function isUnavailable(directive) {
+  const parts = directive.split(':');
+
+  if (parts.length <= 1 || parts[0] !== UNAVAILABLE_AFTER) {
+    return false;
+  }
+
+  const date = Date.parse(parts[1]);
+
+  return !isNaN(date) && date < Date.now();
+}
+
+/**
+ * Returns false if any of provided directives blocks page from being indexed
+ * @param {string} directives
+ * @returns {boolean}
+ */
+function hasBlockingDirective(directives) {
+  return directives.split(',')
+    .map(d => d.toLowerCase().trim())
+    .some(d => BLOCKLIST.has(d) || isUnavailable(d));
+}
+
+/**
+ * Returns false if robots header specifies user agent (e.g. googlebot)
+ * @param {string} directives
+ * @returns {boolean}
+ */
+function hasUA(directives) {
+  const parts = directives.split(':');
+  return parts.length > 1 && parts[0] !== UNAVAILABLE_AFTER;
+}
+
+class IsCrawlable extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      name: 'is-crawlable',
+      description: 'Page isn’t blocked from indexing',
+      failureDescription: 'Page is blocked from indexing',
+      helpText: 'The "Robots" directives tell crawlers how your content should be indexed. ' +
+      '[Learn more](https://developers.google.com/search/reference/robots_meta_tag).',
+      requiredArtifacts: ['MetaRobots'],
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    return artifacts.requestMainResource(artifacts.devtoolsLogs[Audit.DEFAULT_PASS])
+      .then(mainResource => {
+        const blockingDirectives = [];
+
+        if (artifacts.MetaRobots) {
+          const isBlocking = hasBlockingDirective(artifacts.MetaRobots);
+
+          if (isBlocking) {
+            blockingDirectives.push({
+              source: {
+                type: 'node',
+                snippet: `<meta name="robots" content="${artifacts.MetaRobots}" />`,
+              },
+            });
+          }
+        }
+
+        mainResource.responseHeaders
+          .filter(h => h.name.toLowerCase() === ROBOTS_HEADER && !hasUA(h.value) &&
```

Comment:
so to be clear, we're looking for the robots header that has a user agent specified *and* is specified to block?

seems like this might miss a few cases maybe I'm misunderstanding

can there not be multiple directives in the header?

Reply:
we are looking for robots header that *doesn't* have UA specified (we don't support UA specific headers) and is blocking indexing